### DISCUSSION TITLE
[Help]: I was wondering it was possible to start the carousel centered

### DISCUSSION THREAD

**Discussion by leebaroneau** [davidjerleke/embla-carousel#1038] [Help] [answered]
### Summary

I was wondering if it was possible to start the carousel items centred or can it only start with the first item at the start, center or end? 

I have the slider like this and the distance of the controls are very wide when there are less items in the carousel.
https://hardwarebox.com.au/collections/entrance-front-door-handles
![image](https://github.com/user-attachments/assets/cb3b395e-bfc1-4e78-b4c9-4506c25aa9b5)

I'm trying to get something like this or the controls on the outside contract in depending on the number of the items in the slider.
![image](https://github.com/user-attachments/assets/bb64d889-8316-41f7-930f-ef35c1ef51aa)

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [X] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Additional information

```js
I'm using the CDN method with the following setup for the code.
 <script src="https://unpkg.com/embla-carousel/embla-carousel.umd.js"></script>
  <script src="https://unpkg.com/embla-carousel-autoplay/embla-carousel-autoplay.umd.js"></script>
  <script>
    // Grab wrapper nodes
    const emblaNode = document.querySelector('.embla');

    // Grab button nodes
    const prevButtonNode = emblaNode.querySelector('.embla__button--prev');
    const nextButtonNode = emblaNode.querySelector('.embla__button--next');

    const options = {
      slidesToScroll: 'auto',
    };

    const plugins = [EmblaCarouselAutoplay()];
    // Initialize the carousel
    const emblaApi = EmblaCarousel(emblaNode, options, plugins);

    // Add click listeners
    prevButtonNode.addEventListener('click', emblaApi.scrollPrev, true);
    nextButtonNode.addEventListener('click', emblaApi.scrollNext, true);
  </script>
```

**Comment by davidjerleke** [comment]
Hi @leebaroneau,

Thanks for your question. 

- You can write your custom alignment:

```tsx
const options = {
  slidesToScroll: 'auto',
  align: (viewSize, snapSize, index) => {
    // return your own alignment which should be a number;
  }
};
```

- However, another approach would be to have your viewport element centered and not as wide as the screen.
- Another approach would be to wrap every four slide in a div.

**Reply by leebaroneau** [reply]
Thank you for the suggestion.

I tried it here with a codepen [here ](https://codepen.io/leebaroneau/pen/eYqRBrX)without any luck. Did I have the API incorrectly setup?

**Comment by sarussss** [comment]
Hi @leebaroneau,
You must set [containScroll](https://www.embla-carousel.com/api/options/#containscroll) is false. to be able to use align option.
You can check [demo](https://codesandbox.io/p/sandbox/embla-carousel-generator-vanilla-forked-fy8r26?file=%2Fsrc%2Fjs%2Findex.js%3A14%2C15).

**Comment by leebaroneau** [comment, marked as answer]
My solution was to calculate an offset based on the number of slides in the slider.

```tsx
const options = {
  containScroll: false,
  loop: true,
  align: (viewSize, snapSize, index) => {
    console.log(viewSize, snapSize, index);
    let slideElement = document.querySelector('.embla__slide');
    let slideCount = document.querySelectorAll('.embla__slide').length;
    let slideWidth = slideElement.offsetWidth;
    let slideNumber = viewSize / slideWidth;

    if (slideCount < slideNumber) {
      // Get all elements with the class 'embla__button'
      const buttons = document.querySelectorAll('.embla__button');

      // Iterate over each button and set its display style to 'none'
      buttons.forEach(button => {
        button.style.display = 'none';
      });
      console.log(slideCount, slideNumber, slideWidth);
      let offsetSlider = ((slideNumber - slideCount) * slideWidth) / 2;
      console.log(offsetSlider);
      return offsetSlider;
    } else {
      return 0;
    }
  },
};

```
